### PR TITLE
EVEREST-785: fix validation when creating a monitoring instance

### PR DESCRIPTION
**Pull request by michal-kralik**
[EVEREST-785](https://jira.percona.com/browse/EVEREST-785) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
EVEREST-785

Monitoring instance was not properly configured when username was empty.

**Cause:**
API validation did not properly check input parameters.

**Solution:**
Updated validation

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

[EVEREST-785]: https://perconadev.atlassian.net/browse/EVEREST-785?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/validation.go [hunk 1]
```diff
@@ -350,8 +350,10 @@ func validateCreateMonitoringInstanceRequest(ctx echo.Context) (*CreateMonitorin
 			return nil, fmt.Errorf("pmm key is required for type %s", params.Type)
 		}
 
-		if params.Pmm.ApiKey == "" && params.Pmm.User == "" && params.Pmm.Password == "" {
-			return nil, errors.New("one of pmm.apiKey, pmm.user or pmm.password fields is required")
+		if params.Pmm.ApiKey == "" &&
+			((params.Pmm.User == "" && params.Pmm.Password == "") ||
+				(params.Pmm.User == "" && params.Pmm.Password != "")) {
```
Comment (recharte) [inline api/validation.go:355]: Should we also have the case when the User is set and the Password is not?

Comment (recharte) [reply api/validation.go:355]: Ok, I didn't know that an empty password was a valid one. Does PMM accept it?

File: api/validation.go [hunk 2]
```diff
@@ -350,8 +350,10 @@ func validateCreateMonitoringInstanceRequest(ctx echo.Context) (*CreateMonitorin
 			return nil, fmt.Errorf("pmm key is required for type %s", params.Type)
 		}
 
-		if params.Pmm.ApiKey == "" && params.Pmm.User == "" && params.Pmm.Password == "" {
-			return nil, errors.New("one of pmm.apiKey, pmm.user or pmm.password fields is required")
```
Comment (oksana-grishchenko) [inline api/validation.go:354]: Looks like the enclosing function is a good candidate to cover it with unit tests. Up to you though, it's not a blocker

**Review by oksana-grishchenko — APPROVED** [review]

File: api/validation.go (hunk 1 above)
Comment (recharte) [reply api/validation.go:355]: I checked and PMM doesn't allow an empty password so we might as well enforce it.

Comment (michal-kralik) [reply api/validation.go]: I changed this to make password required

**Review by recharte — APPROVED** [review]

File: api/validation.go (hunk 2 above)
Comment (michal-kralik) [reply api/validation.go:354]: yes, that's a good point. I think the code would benefit from a better unit test coverage. We can consider dedicating some time for it EVEREST-792

**Pull request merged**